naresh/src/main/java/com/lftechnology/phpjava/ems/utlis/Router.java

```diff
@@ -0,0 +1,112 @@
+package com.lftechnology.phpjava.ems.utlis;
+
+import com.lftechnology.phpjava.ems.constants.Constant;
+import com.lftechnology.phpjava.ems.controllers.EmployeeController;
+import com.lftechnology.phpjava.ems.controllers.LoginController;
+import com.lftechnology.phpjava.ems.services.EmployeeService;
+import com.lftechnology.phpjava.ems.views.CommonViewUtility;
+
+import java.sql.SQLException;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * RouterService
+ *
+ * @author Naresh Maharjan <nareshmaharjan@lftechnology.com>
+ * @since August, 09 2016
+ */
+public class Router {
+
+    private static LoginController loginController = new LoginController();
+    private static EmployeeController employeeController = new EmployeeController();
+    private static CommonUtility commonUtility = new CommonUtility();
+
+    static {
+        try {
+            onRouteBootstrap();
+        } catch (SQLException e) {
+            e.printStackTrace();
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    /**
+     * Method to be executed on router startup.
+     * Will invoke checkAndCreateAdminUser() method
+     * to create admin user if the system doesn't have one.
+     *
+     * @author Naresh Maharjan <nareshmaharjan@lftechnology.com>
+     */
+    public static void onRouteBootstrap() throws Exception {
+
+        SqlRunner sqlRunner = new SqlRunner();
+        sqlRunner.executeSql();
+
+        EmployeeService employeeService = new EmployeeService();
+        employeeService.checkAndCreateAdminUser();
+    }
+
+    /**
+     * Show Welcome Screen
+     *
+     * @author Naresh Maharjan <nareshmaharjan@lftechnology.com>
+     */
+    public static void showWelcomeScreen() {
+
+        CommonViewUtility.showWelcomeScreen();
+    }
+
+    /**
+     * @author Naresh Maharjan <nareshmaharjan@lftechnology.com>
+     */
+    public static void showLoginMenuAndLoginToSystem() {
+        if (!commonUtility.getUserService().isUserLoggedIn()) {
+            showWelcomeScreen();
+            loginController.login();
+            commonUtility = (CommonUtility) loginController.getData().get(Constant.COMMON_UTILITY);
+        }
+    }
+
+    /**
+     * @author Naresh Maharjan <nareshmaharjan@lftechnology.com>
+     */
+    public static void showMenu() {
+        showLoginMenuAndLoginToSystem();
+        if (!commonUtility.isLoggedIn()) {
+            CommonViewUtility.showMessageAndContinue(Constant.INVALID_CREDENTAILS);
+        } else {
+            Map<String, Object> commonUtilityMap = new HashMap<>();
+            commonUtilityMap.put(Constant.COMMON_UTILITY, commonUtility);
+            employeeController.setData(commonUtilityMap);
+            employeeController.postLoginScreen();
+        }
+    }
+
+    /**
+     * @param menu
+     * @author Naresh Maharjan <nareshmaharjan@lftechnology.com>
+     */
+    public static void initiateInitialMenuAction(int menu) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The Cyclomatic Complexity of this method "initiateInitialMenuAction" is 14 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar.java.lftechnology.com:9000/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)